src/shared/use-server-time-as-local/use-server-date-time-as-local.js

```diff
@@ -0,0 +1,21 @@
+import { useConfig } from '@dhis2/app-runtime'
+
+export const useServerDateTimeAsLocal = (dateTime) => {
+    const { systemInfo } = useConfig()
+
+    if (!dateTime) {
+        return undefined
+    }
+
+    const localNow = new Date()
+    localNow.setMilliseconds(0)
+    const nowAtServerTimeZone = new Date(
+        localNow.toLocaleString('en-US', {
+            timeZone: systemInfo.serverTimeZoneId,
+        })
+    )
+    const timestamp = new Date(dateTime).getTime()
+    const timeOffset = localNow.getTime() - nowAtServerTimeZone.getTime()
+
+    return new Date(timestamp + timeOffset)
```

Comment:
I think this should be `return new Date(timestamp - timeOffset)`

above you are calculating `offset = local - server`
then in this line you are creating the new date as `local + offset` or `local + (local - server)`
if you make this subtraction, it will be `local - (local - server) = server`

(Alternately, you could change the offset calculation to be `offset = server - local`)

Reply:
God, I just realised I got all of this completely the wrong way around. The calculation I was doing in the other app was the inverse of what we're trying to do here. Thanks for pointing that out.